src/ipc-handlers.ts

```diff
@@ -862,49 +862,71 @@ export async function getThumbnailData( _event: IpcMainInvokeEvent, id: string )
 	return getImageData( path );
 }
 
-export function openTerminalAtPath(
+async function checkPowerShellAvailability() {
+	if ( process.platform !== 'win32' ) {
+		return false;
+	}
+
+	return new Promise( ( resolve ) => {
+		exec( 'powershell -Command "Get-Host"', ( error, stdout, stderr ) => {
+			resolve( ! error && ! stderr );
+		} );
+	} );
+}
+
+export async function openTerminalAtPath(
 	_event: IpcMainInvokeEvent,
 	targetPath: string,
 	{ wpCliEnabled }: { wpCliEnabled?: boolean } = {}
 ) {
-	return new Promise< void >( ( resolve, reject ) => {
-		const platform = process.platform;
-		const cliPath = nodePath.join( getResourcesPath(), 'bin' );
-
-		const exePath = app.getPath( 'exe' );
-		const appDirectory = app.getAppPath();
-		const appPath = ! app.isPackaged ? `${ exePath } ${ appDirectory }` : exePath;
-
-		let command: string;
-		if ( platform === 'win32' ) {
-			// Windows
-			if ( wpCliEnabled ) {
-				command = `start cmd /K "set PATH=${ cliPath };%PATH% && set STUDIO_APP_PATH=${ appPath } && cd /d ${ targetPath }"`;
-			} else {
-				command = `start cmd /K "cd /d ${ targetPath }"`;
-			}
-		} else if ( platform === 'darwin' ) {
-			// macOS
-			const loadWpCliCommand = `clear && export PATH=\\"${ cliPath }\\":$PATH && export STUDIO_APP_PATH=\\"${ appPath }\\" &&`;
-			const script = `
-			tell application "Terminal"
-				if not application "Terminal" is running then launch
-				do script "${ wpCliEnabled ? loadWpCliCommand : '' } cd ${ targetPath } && clear"
-				activate
-			end tell`;
-			command = `osascript -e '${ script }'`;
-		} else if ( platform === 'linux' ) {
-			// Linux
-			if ( wpCliEnabled ) {
-				command = `export PATH=${ cliPath }:$PATH && export STUDIO_APP_PATH="${ appPath }" && gnome-terminal -- bash -c 'cd ${ targetPath }; exec bash'`;
-			} else {
-				command = `gnome-terminal --working-directory=${ targetPath }`;
-			}
+	const platform = process.platform;
+	const cliPath = nodePath.join( getResourcesPath(), 'bin' );
+
+	const exePath = app.getPath( 'exe' );
+	const appDirectory = app.getAppPath();
+	const appPath = ! app.isPackaged ? `${ exePath } ${ appDirectory }` : exePath;
+
+	let command: string;
+	if ( platform === 'win32' ) {
+		// Windows
+		const isPowerShellAvailable = await checkPowerShellAvailability();
+		const defaultShell = process.env.ComSpec || 'cmd.exe';
```

Comment:
This mirrors Node's internal logic for locating the default Windows shell. See https://nodejs.org/api/child_process.html#default-windows-shell